### PR TITLE
Added `make install-dev` and a stronger `make clean` for easier dev on-boarding and release upgrades.

### PR DESCRIPTION
**Pull request by dmoore247**
… easier on-boarding experience

The problem was `make test` was reporting incompatible versions. The `hatch` environment needed cleaning up. Also added `make install-dev` as a convention to install the developer environment (hatch).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,13 @@
+all: clean lint fmt test
+
 clean:
 	rm -fr htmlcov .mypy_cache .pytest_cache .ruff_cache .coverage coverage.xml
+	hatch env remove unit
+
+install-dev:
```
Comment (nfx) [inline Makefile:7]:
```suggestion
dev:
```
We use this for python sdk

File: Makefile [hunk 2]
```diff
@@ -11,4 +19,5 @@ test:
 	hatch run unit:test
 
 test-cov:
```
Comment (nfx) [inline Makefile:21]: Perhaps call it "coverage"

**Review by nfx — CHANGES_REQUESTED** [review]
Can you also modify the contributing.md?

File: Makefile [hunk 3]
```diff
@@ -1,5 +1,13 @@
+all: clean lint fmt test
+
 clean:
 	rm -fr htmlcov .mypy_cache .pytest_cache .ruff_cache .coverage coverage.xml
+	hatch env remove unit
+
+install-dev:
+	pip install hatch
+	hatch env create
+	hatch run pip install -e '.[test]'
```
Comment (nfx) [inline Makefile:10]: Can you output the path to python interpreter here?

**Review by nfx — CHANGES_REQUESTED** [review]
Please also modify the contributing.md

File: Makefile (hunk 3 above)
Comment (dmoore247) [reply Makefile:10]: as in `which python` ?

**Review by nfx — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (larsgeorge-db) [reply Makefile:10]: I think Serge means this:
```
hatch run python -c "import sys; print(sys.executable)"
```

**Pull request merged**